### PR TITLE
Added gen_html_with_total_len to generating html taking in account ta…

### PR DESCRIPTION
**Pull request by renzon**
…g chars

### CODE REVIEW

**Comment by rochacbruno** [comment]
@omaciel do you want to keep python 2.6 support here?

**Comment by renzon** [comment]
@rochacbruno Good question. I took a look on travis and the reason it's is failing on 2.6 is OderedDict use. So it's not related to this change

**Comment by rochacbruno** [comment]
@omaciel @renzon I think support for Python 2.6 can be dropped everywhere :)

**Comment by omaciel** [comment]
@rochacbruno @renzon the only reason I kept python 2.6 was because of @jlaska since he was using it for Ansible. @jlaska is this still the case?

File: fauxfactory/__init__.py [hunk 1]
```diff
@@ -810,6 +811,32 @@ def gen_html(length=10):
...
+    """Returns a random string made up of html characters.
+    This differ from fauxfactory.gen_html because length takes html tag chars
+    into account
+    :param int length: Length for random data.
+    :returns: A random string made up of html characters.
+    :rtype: str
+
+    """
+    if length < 8:
+        raise ValueError(u'Impossible generate html with len less then 7')
```
Comment (omaciel) [inline fauxfactory/__init__.py]: "Impossible **to** generate **HTML** with ~~len~~ less than 7 **characters**"?

File: fauxfactory/__init__.py [hunk 2]
```diff
@@ -810,6 +811,32 @@ def gen_html(length=10):
...
+
+    random.seed()
+    html_tag = random.choice(HTML_TAGS)
+    maybe_len = length - (len(html_tag) * 2 + 5)
```
Comment (omaciel) [inline fauxfactory/__init__.py]: Can you comment on the logic behind the calculation?

**Comment by jlaska** [comment]
> @rochacbruno @renzon the only reason I kept python 2.6 was because of @jlaska since he was using it for Ansible. @jlaska is this still the case?

I don't believe we have any requirements for running `py.test` on el6 environments anymore.  Thanks for reaching out!

**Comment by rochacbruno** [comment]
If so, the way to go is adding the dependency https://pypi.python.org/pypi/ordereddict and make it conditional to Py2.6

**Comment by omaciel** [comment]
@rochacbruno I'm all in favor of dropping py2.6 now then :)

**Comment by rochacbruno** [comment]
@renzon Lets drop 2.6? push in your branch removing the 2.6 from tox.ini and .travia files.

File: fauxfactory/__init__.py [hunk 3]
```diff
@@ -543,11 +551,13 @@ def gen_ipaddr(ip3=False, ipv6=False, prefix=()):
     :param bool ip3: Whether to generate a 3 or 4 group IP.
     :param bool ipv6: Whether to generate IPv6 or IPv4
     :param list prefix: A prefix to be used for an IP (e.g. [10, 0, 1]). It
-        must be an iterable with strings or integers. Can be left unspecified or
+        must be an iterable with strings or integers. Can be left
+        unspecified or
         empty.
```
Comment (omaciel) [inline fauxfactory/__init__.py]: I think you can move this line up?

File: fauxfactory/__init__.py [hunk 4]
```diff
@@ -810,6 +820,32 @@ def gen_html(length=10):
...
+    """Returns a random string made up of html characters.
+    This differ from fauxfactory.gen_html because length takes html tag chars
+    into account
+    :param int length: Length for random data.
+    :returns: A random string made up of html characters.
+    :rtype: str
+
+    """
+    if length < 8:
+        raise ValueError(u'Impossible generate html with less than 7 chars')
```
Comment (omaciel) [inline fauxfactory/__init__.py]: Impossible **to** generate :)

File: tox.ini [hunk 5]
```diff
@@ -2,7 +2,7 @@
 minversion = 2.6
```
Comment (rochacbruno) [inline tox.ini]: This should be 2.7

File: fauxfactory/__init__.py [hunk 6]
```diff
@@ -634,8 +642,8 @@ def gen_mac(delimiter=':', multicast=None, locally=None):
 
     octets = [first_octet]
     octets.extend([
-        random.randint(0, 255) for _ in range(5)
-    ])
+                      random.randint(0, 255) for _ in range(5)
+                  ])
```
Comment (rochacbruno) [inline fauxfactory/__init__.py]: Is the indentation ok here?

Comment (renzon) [inline fauxfactory/__init__.py]: Sometimes Pycharm reformatting sucks. But at least I put the code in one line only and without using list.

**Pull request merged**